### ISSUE TITLE
Enhancement: consistent-type-imports Fix type imports used as values

### ISSUE THREAD

**Issue by eliasm307** [typescript-eslint/typescript-eslint#11527]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+plugin+rule+option%22) and found none that match my proposal.
- [x] I have searched the [current rule list](https://typescript-eslint.io/rules/#rules) and found no rules that match my proposal.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [x] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Link to the rule's documentation

https://typescript-eslint.io/rules/consistent-type-imports/

### Description

The `consistent-type-imports` includes a `fixStyle` option which is for converting a value import into a type import "when an import is detected as used only in the type position", however, the rule does not check/fix the opposite way ie when a type import is detected as used NOT only in the type position.

This is not ideal because the rule can create some awkward DX for things that can be used as both types and values depending on how the code changes over time e.g. imagine someone is writing code and they start with code like

```ts
import {SomeEnum, SomeClass} from 'foo'

let enum1: SomeEnum | undefined
let instance1: SomeClass | undefined
```

at this point if the rule runs it will auto fix the imports to be type imports (e.g. `import {type SomeEnum, type SomeClass} from 'foo'`), however if more code is written and it becomes:

```ts
import {type SomeEnum, type SomeClass} from 'foo'

let enum1: SomeEnum | undefined
let instance1: SomeClass | undefined

if(someCondition) {
  enum1 = SomeEnum.Value
  instance1 = new SomeClass()
}
```

if the rule runs again now it will do nothing, however there would be a ts issue as type only inputs are used as values, however the dev did not add those type imports, the lint rule did, but its up to the dev to manually fix this case which is inconvenient. 

Ideally, the rule should also fix the code to remove the type imports where necessary and handle cases where only some named imports need to be converted e.g.

```ts
import type {UsedAsType, UsedAsValue} from 'foo'

const v: UsedAsType = UsedAsValue;
```

### Fail

```TypeScript
import {type UsedAsType, type UsedAsValue} from 'foo'

const v: UsedAsType = UsedAsValue;
```

### Pass

```TypeScript
import {type UsedAsType, UsedAsValue} from 'foo'

const v: UsedAsType = UsedAsValue;
```

**Comment by bradzacher** [comment]
This has been asked countless times please use the search. 

The rule is working as intended. Use this rule to enforce a specific specifier style.

https://github.com/import-js/eslint-plugin-import/blob/main/docs%2Frules%2Fconsistent-type-specifier-style.md

**Comment by eliasm307** [comment]
@bradzacher thanks for the response, however i did use the search and couldnt find issues similar to what I'm meaning, I have checked again, the rule currently does not fix type imports used as values from what I see and i've not seen an issue discussing a request like this? 

also the `import/consistent-type-specifier-style` rule you linked does not seem related to what the issue is about? it does not seem that rule converts type imports to value imports? it is just about different styles of type imports? I'm not asking about the specifier style btw, i mean converting type imports to value imports

incase its not clear i'm talking about this issue 

![Example of a type import used as a value causing a TS issue](https://github.com/user-attachments/assets/24c72341-66bb-4bc4-ad90-d77769fa7817)

ie the rule fixes imports to be type imports but cannot fix them the other way, so in some cases the rule just creates work if it "fixes" wip code but the final code uses those imports as values

https://typescript-eslint.io/play/#ts=5.1.6&showAST=scope&fileType=.ts&code=JYWwDg9gTgLgBAbzjAnmApnAqgZ3QEwEEcAVNdAGmXOzyJwDUBDAGwFdMBfOAMyghBwA5DwgQhAWABQ0gMYQAdjngA3AFy0CxMhjgBeTfWbt0AbiA&eslintrc=N4KABGBEBOCuA2BTAzpAXGUEKQAIBcBPABxQGNoBLY-AWhXkoDt8B6MgeyeUuX0Ra0ipWpQC2xDtHyoMAbUiJo0KZAA04bBGBRi0RADMl6KMMSiJUmeqgHKADwDKRJCcjNGTc2YuTpqMABfTQgAXU1gwKA&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

Please let me know if i'm missing something and this behaviour already exists somewhere

**Comment by bradzacher** [comment]
Apologies I pattern matched your OP and got it wrong.

Regardless this is still working as intended. The rule does not remove the modifier, it only checks for _missing_ modifiers and adds them if they are missing.

If you want to use an imported type as a value then you must manually remove the type modifier from the import specifier.

**Comment by eliasm307** [comment]
just to clarify, since the rule has a way to determine if an import is only used as a type, i'm saying use the same condition to check existing type imports and if they dont meet the condition the import is converted to a value import

this could be a new option? I can work on this if required because the rule is a bit inconvenient in some cases, and as you say

> you must manually remove the type modifier from the import specifier.

for me its annoying to have to manually fix changes made by a lint rule

**Comment by eliasm307** [comment]
@bradzacher  if you dont want it to be part of this rule then can it be a new rule?

however imo i think it should be in this rule as it already has similar behaviour in the opposite way and to avoid behaviour mismatches ie if this rule thinks an import should be a type and another rule having the opposite result

**Comment by bradzacher** [comment]
I don't see how this makes sense as a lint rule in a general purpose plugin like ours.
The current rule is about stylistic correctness -- enforcing that the modifier is used when it makes sense to. Generally speaking this does much, much, much more regularly apply to types and not to dual value/type constructs (classes and enums). So it's a "sometimes necessary" minor pain point in the developer workflow of using the _optional_ type modifier on imports

Additionally TS already has a quick fixer built in already for this:
<img width="707" height="171" alt="screenshot of quick fix" src="https://github.com/user-attachments/assets/67a85fe8-a08a-4376-9b16-212d3fc05a7a" />

Finally this rule is 5 years old and is a very highly used rule (despite it not being in our recommended set!) and this is the first time someone has mentioned this frustration. Which suggests that people generally don't feel burdened bythis.

If you want to build this rule and publish it as a 3rd party plugin -- please feel free! But I do not believe it is broadly applicable enough or high-value enough to live within our plugin.

**Comment by eliasm307** [comment]
fair enough, thanks for the feedback
